central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -198,6 +232,38 @@ func (m *managerImpl) ProcessScanRequest(ctx context.Context, scanRequest *stora
 	return scanRequest, nil
 }
 
+func (m *managerImpl) processClusterDelete(ctx context.Context, scanRequest *storage.ComplianceOperatorScanConfigurationV2, clusters []string) {
+	for _, clusterID := range clusters {
+		log.Infof("SHREWS -- trying to delete cluster %q", clusterID)
+		// id for the request message to sensor
+		sensorRequestID := uuid.NewV4().String()
+
+		sensorMessage := &central.MsgToSensor{
```

Comment:
What happens if a delete and a scan are executed at the same time?
The scan requests are tracked in `runningRequests`, but the `processClusterDelete` is not.

Reply:
Yeah that is a potential race condition that would be extremely unlikely.  But if it occurred we would then track the cluster with the scan configuration again even though the objects would have been removed from sensor.  So we would be in an inconsistent state.  Thus we should probably remove any such `runningRequests` when we `processClusterDelete`

Reply:
In production it is unlikely. In a dev system I am seeing me running into this quite often.

Reply:
> Thus we should probably remove any such runningRequests when we processClusterDelete

This sounds like the solution. Imho can be done in a second PR if it is faster.

Reply:
Will do separately.  For tracking
https://issues.redhat.com/browse/ROX-22509